cfme/tests/control/test_actions.py

```diff
@@ -31,188 +29,109 @@
-
-    def __getattr__(self, key):
-        """Creates partial functions proxying to mgmt_system.<function_name>(vm_name)"""
-        func = getattr(self._prov.mgmt, key)
-        return partial(func, self._vm)
-
-
-def get_vm_object(appliance, vm_name):
-    """Looks up the CFME database for the VM.
-
-    Args:
-        vm_name: VM name
-
-    Returns:
-        If found, returns a REST object
-        If not, `None`
-    """
-    try:
-        return appliance.rest_api.collections.vms.find_by(name=vm_name)[0]
-    except IndexError:
-        return None
-
-
-@pytest.fixture(scope="module")
-def vm_name(provider):
-    return random_vm_name("action", max_length=16)
-
-
-def _get_vm(request, provider, template_name, vm_name):
+def _get_vm(request, provider, template_name):
+    vm_name = random_vm_name("action", max_length=16)
     if provider.one_of(RHEVMProvider):
         kwargs = {"cluster": provider.data["default_cluster"]}
     elif provider.one_of(OpenStackProvider):
         kwargs = {}
-        if 'small_template' in provider.data.templates:
-            kwargs = {"flavor_name": provider.data.provisioning.get('instance_type')}
+        if "small_template" in provider.data.templates:
+            kwargs = {"flavor_name": provider.data.provisioning.get("instance_type")}
     elif provider.one_of(SCVMMProvider):
         kwargs = {
-            "host_group": provider.data.get("provisioning", {}).get("host_group", "All Hosts")}
+            "host_group": provider.data.get("provisioning", {}).get("host_group", "All Hosts")
+        }
     else:
         kwargs = {}
 
     collection = provider.appliance.provider_based_collection(provider)
     vm = collection.instantiate(vm_name, provider, template_name)
-
-    try:
-        deploy_template(
-            provider.key,
-            vm_name,
-            template_name=template_name,
-            allow_skip="default",
-            power_on=True,
-            **kwargs
-        )
-    except TimedOutError:
-        message = "{} failed provisioning, check its status!".format(provider.key)
-        logger.exception(message)
-        try:
-            vm.cleanup_on_provider()
-        except TimedOutError:
-            logger.error("Could not delete VM %s!", vm_name)
-    finally:
-        # If this happened, we should skip all tests from this provider in this module
-        pytest.skip(message)
-
+    vm.create_on_provider(delete_on_failure=True, find_in_cfme=True, **kwargs)
     request.addfinalizer(vm.cleanup_on_provider)
-
-    # Make it appear in the provider
     provider.refresh_provider_relationships()
 
-    # Get the REST API object
-    api = wait_for(
-        get_vm_object,
-        func_args=[provider.appliance, vm_name],
-        message="VM object {} appears in CFME".format(vm),
-        fail_condition=None,
-        num_sec=600,
-        delay=15,
-    )[0]
-
-    return VMWrapper(provider, vm_name, api)
+    return vm
 
 
-@pytest.fixture(scope="module")
```

Comment:
@digitronik These tests were failing in part due to the fact that they were using the same VM for each test. If a test fails, then the state of the VM got messed up and a bunch of tests would fail.